lldb/source/Plugins/ExpressionParser/Swift/SwiftREPL.cpp

```diff
@@ -542,7 +542,7 @@ int SwiftREPL::CompleteCode(const std::string &current_code,
   // our own copy of the AST and using this separate AST for completion.
   //----------------------------------------------------------------------
   Status error;
-  if (!m_swift_ast_sp) {
+  if (!m_swift_ast) {
```

Comment:
here we also need to acquire the scratch context lock. That's very easy if you just call target.GetSwiftScratchContext().